getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,152 @@
+
+val storage: Storage = StorageOptions.getDefaultInstance().service
+val bucket = storage.get(System.getenv("GOOGLE_STORAGE_BUCKET"))
+
+const val RESOURCES_PATH = "resources/" // Make sure to adapt path if running locally
+const val EMOJIS_PATH = RESOURCES_PATH + "emojis/"
+
+enum class Emoji {
+    JOY, ANGER, SURPRISE, SORROW, NONE
+}
+
+val emojiPic = mapOf(
+    Emoji.JOY to "joy.png",
+    Emoji.ANGER to "anger.png",
+    Emoji.SURPRISE to "surprise.png",
+    Emoji.SORROW to "sorrow.png",
+    Emoji.NONE to "none.png"
+)
+
+// Returns best emoji based on detected emotions likelihoods
+fun bestEmoji(annotation: FaceAnnotation): Emoji {
+    val emotionsLikelihood: Array<Likelihood> = arrayOf(Likelihood.VERY_LIKELY, Likelihood.LIKELY, Likelihood.POSSIBLE)
+    val emotions = mapOf(
+        Emoji.JOY to annotation.joyLikelihood,
+        Emoji.ANGER to annotation.angerLikelihood,
+        Emoji.SURPRISE to annotation.surpriseLikelihood,
+        Emoji.SORROW to annotation.sorrowLikelihood
+    )
+
+    for (likelihood in emotionsLikelihood) { // In this order: VERY_LIKELY, LIKELY, POSSIBLE
+        for (emotion in emotions) { // In this order: JOY, ANGER, SURPRISE, SORROW
+            if (emotion.value == likelihood) return emotion.key // Returns emotion corresponding to likelihood
+        }
+    }
+    return Emoji.NONE
+}
+
+class GcsObject(val bucketName: String, val blobName: String)
+class EmojifyResponse(val original: GcsObject, val emojified: GcsObject, val emojifiedUrl: String)
+
+@RestController
+class EmojifyController {
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): EmojifyResponse? {
+
+        val publicUrl: String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+        val log = Logger.getLogger(Application::class.java.name)
+
+        // Setting up image annotation request
+        val requests = ArrayList<AnnotateImageRequest>()
+        val vision = ImageAnnotatorClient.create()
+        val source = ImageSource.newBuilder().setGcsImageUri("gs://${bucket.name}/$objectName").build()
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
+        vision.close()
+
+        for (resp in response.responsesList) {
+            if (resp.hasError()) {
+                log.severe(resp.error.message)
+                return null
+            }
+            // Downloads source image to /tmp/<objectName>
+            val originalBlob = bucket.get(objectName)
+            val localFilePath = Paths.get("/tmp/$objectName")
```

Comment:
This should use:
1. `File.createTempFile()` to create a temporary file. 
2. Paths is prob not necessary here. But if used, use the vargs get, e.g., Paths.get("tmp", objectName). This is because the path separator differs between OS.